src/pages/index.js

```diff
@@ -1,89 +1,241 @@
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithImage from "../components/PopupWithImage.js";
+import PopupWithDelete from "../components/PopupWithDelete.js";
+import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
+import Section from "../components/Section.js";
+import Card from "../components/Card.js";
+import Api from "../components/Api.js";
+import "./index.css";
+import { validationSettings, variables } from "../utils/constants.js";
 
-import{
-  initialCards,
-  addNewModalCard,
-  profileEditModal,
-  cardListEl,
-  validationSettings,
-  profileTitleInput,
-  profileDescriptionInput,
-  profileEditButton,
-  addNewCardButton,
-} from "../utils/constants.js";
+// API
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f00dbe4d-3bcf-40e3-a46f-9e1ed8206bd5",
+    "Content-Type": "application/json",
+  },
+});
+
+// SECTION
+
+let section;
+
+api
+  .getInitialCards()
+  .then((res) => {
+    section = new Section(
+      {
+        items: res,
+        renderer: (cardData) => {
+          createCard(cardData);
+        },
+      },
+      "cards__list"
+    );
+    section.renderItems();
+    console.log(res);
+  })
+  .catch(console.error);
+
+api
+  .loadInfo()
+  .then((res) => {
+    console.log(res);
+    userInfo.setUserInfo(res.name, res.about);
+    userInfo.setAvatar(res.avatar);
+  })
+  .catch(console.error);
+
+// POPUP WITH IMAGE
+
+const popupWithImage = new PopupWithImage(
+  "#modal-preview");
+popupWithImage.setEventListeners();
+
+// SAVE BUTTON TEXT CHANGE FUNCTION FOR FORMS
+
+function setButtonText(button, text) {
+  button.textContent = text;
+}
+
+// CARD
+
+function handleImageClick(cardData) {
+  popupWithImage.open(cardData);
+}
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  section.addItem(card.generateCard());
+}
+
+function handleAddCardSubmit(inputValues) {
+  setButtonText(variables.addModalButton, "Saving...");
+  api
+    .addNewCard(inputValues)
+    .then((res) => {
+      console.log(res);
+      // create a new card
+      createCard(res);
+      // render it
+      variables.profileAddForm.reset();
+      // toggle button state
+      addFormValidator.toggleButtonState();
+      // close modal
+      addPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(variables.addModalButton, "Save");
+    });
+}
+
+// HANDLE DELETE MODAL
+
+const deleteModal = new PopupWithDelete(
+  variables.deleteCard);
+deleteModal.setEventListeners();
+
+function handleDeleteClick(card) {
+  setButtonText(deleteModal.close, "Loading...");
+  // open the modal
+  deleteModal.open();
+  // set the submit action
+  deleteModal.setSubmitAction(() => {
+    api
+      .deleteCard(card._id)
```

Comment:
Needs correction: don't forget to notify the user about loading process by changing the button text (like you do that for other forms using ` setButtonText` function)